samples/DotNet/Microsoft.ServiceBus.Messaging/GeoDR/SBGeoDR2/SBGeoDR2/Program.cs

```diff
@@ -0,0 +1,137 @@
+
+            //// 2. Create Secondary Namespace (optional if you already have an empty namespace available)
+            var namespaceParams2 = new SBNamespace()
+            {
+                Location = "North Central US",
+                Sku = new SBSku()
+                {
+                    Name = SkuName.Premium,
+                    Capacity = 1
+                }
+            };
+
+            //// If you re-run this program while namespaces are still paired this operation will fail with a bad request.
+            //// this is because we block all updates on secondary namespaces once it is paired
+            var namespace2 = client.Namespaces.CreateOrUpdate(resourceGroupName, geoDRSecondaryNS, namespaceParams2);
+
+            // 3. Pair the namespaces to enable DR.            
+            ArmDisasterRecovery drStatus = client.DisasterRecoveryConfigs.CreateOrUpdate(
+                resourceGroupName,
+                geoDRPrimaryNS,
+                alias,
+                new ArmDisasterRecovery { PartnerNamespace = geoDRSecondaryNS });
+
+            while (drStatus.ProvisioningState != ProvisioningStateDR.Succeeded)
+            {
+                Console.WriteLine("Waiting for DR to be setup. Current State: " + drStatus.ProvisioningState);
+
+                drStatus = client.DisasterRecoveryConfigs.Get(
+                    resourceGroupName,
+                    geoDRPrimaryNS,
+                    alias);
+
+                Thread.CurrentThread.Join(TimeSpan.FromSeconds(30));
+            }                        
+
+            client.Topics.CreateOrUpdate(resourceGroupName, geoDRPrimaryNS, "myTopic", new SBTopic());
+            client.Subscriptions.CreateOrUpdate(resourceGroupName, geoDRPrimaryNS, "myTopic", "myTopic-Sub1", new SBSubscription());
+
+            // sleeping to allow metadata to sync across primary and secondary
+            Thread.Sleep(1000 * 60);
+            
+            // 6. Failover. Note that this Failover operations is ALWAYS run against the secondary ( because primary might be down at time of failover )
+            // client.DisasterRecoveryConfigs.FailOver(resourceGroupName, geoDRSecondaryNS, alias);            
+
+            // other possible DR operations
+
+            // 7. Break Pairing
+            // client.DisasterRecoveryConfigs.BreakPairing(resourceGroupName, geoDRPrimaryNS, alias);
+
+            // 8. Delete DR config (alias)
+            // note that this operation needs to run against the namespace that the alias is currently pointing to
+            // client.DisasterRecoveryConfigs.Delete(resourceGroupName, geoDRPrimaryNS, alias);
+        }
+
+        private static string GetAuthorizationHeader()
+        {
+            AuthenticationResult result = null;
+
+            var context = new AuthenticationContext(string.Format("{0}/{1}", activeDirectoryAuthority, tenantId));
+
+            var thread = new Thread(() =>
```

Comment:
Is this truly required?
Could you make this method async and await from the main that could be done async as well?